src/function/FunctionLink.ts

```diff
@@ -298,11 +306,179 @@ export function addLinkTools(
         WorkspaceTerms[WorkspaceElements[WorkspaceLinks[id].source].iri]
           .inScheme
       )
-      ].readOnly;
+    ].readOnly;
   let tools = [verticesTool, segmentsTool];
   if (!readOnly) tools.push(removeButton);
   let toolsView = new joint.dia.ToolsView({
-    tools: tools
+    tools: tools,
   });
   linkView.addTools(toolsView);
 }
+
+export async function getCacheConnections(iri: string) {
+  const terms: { [key: string]: any } = {};
+  const restrictions: { [key: string]: Restriction[] } = {};
+  const subClass: string[] = await fetchSubClasses(
+    AppSettings.contextEndpoint,
+    iri
+  );
+  const subClasses = subClass
+    .filter((subClass) => !(subClass in WorkspaceTerms))
+    .concat(
+      WorkspaceTerms[iri].subClassOf.filter(
+        (subClass) => !(subClass in WorkspaceTerms)
+      )
+    );
+  const relationships: {
+    relation: string;
+    target: string;
+    labels: { [key: string]: string };
+  }[] = [];
+  const connections: {
+    link: string;
+    linkLabels: { [key: string]: string };
+    target: {
+      iri: string;
+      labels: { [key: string]: string };
+      description: { [key: string]: string };
+      vocabulary: string;
+    };
+    direction: string;
+  }[] = [];
+
+  if (AppSettings.representation === Representation.FULL) {
+    await fetchRelationships(AppSettings.contextEndpoint, [iri], restrictions);
```

Comment:
Also do I understand correclty that the `restrictions` array is passed to the `fetchRelationships` method that populates it? It would be much better if the function returned the respective array. In general modifying function arguments is something that should never be done, unless there is a good enough reason for that. It also helps readability.